certsuite/harness.py

```diff
@@ -390,16 +391,18 @@ def run_tests(args, config):
             with adb_b2g.DeviceBackup() as backup:
                 device = backup.device
                 runner = TestRunner(args, config)
-                for suite, groups in runner.iter_suites():
-                    try:
-                        runner.run_suite(suite, groups, log_manager, report_manager)
-                    except:
-                        logger.error("Encountered error:\n%s" %
-                                     traceback.format_exc())
-                        error = True
-                    finally:
-                        backup.restore()
-                        device.reboot()
+                try:
+                    for suite, groups in runner.iter_suites():
+                        try:
+                            runner.run_suite(suite, groups, log_manager, report_manager)
+                        except:
+                            logger.error("Encountered error:\n%s" %
+                                         traceback.format_exc())
+                            error = True
+                finally:
+                    marionette_uninstall()
+                    backup.restore()
+                    device.reboot()
```

Comment:
We should have a discussion for restarting b2g.
IMO, there are 3 options:
1. add option `--restart` to enable restart b2g per test, otherwise the b2g will restart after all tests finished.
2. restart b2g per test.
3. restart b2g after all tests finished.